Examples/Physics_applications/laser_ion/PICMI_inputs_2d.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+from pywarpx import picmi
+
+# Physical constants
+c = picmi.constants.c
+q_e = picmi.constants.q_e
+
+# time-scale with highly kinetic dynamics
+stop_time = 0.2e-12
+
+# Number of cells
+nx = 384
+nz = 512
+
+# Physical domain
+xmin = -7.5e-06
+xmax = 7.5e-06
+zmin = -5.0e-06
+zmax = 25.0e-06
+
+# Domain decomposition
+max_grid_size = 64
+blocking_factor = 32
+
+# Create grid
+grid = picmi.Cartesian2DGrid(
+    number_of_cells=[nx, nz],
+    lower_bound=[xmin, zmin],
+    upper_bound=[xmax, zmax],
+    lower_boundary_conditions=['open', 'open'],
+    upper_boundary_conditions=['open', 'open'],
+    lower_boundary_conditions_particles=['absorbing', 'absorbing'],
+    upper_boundary_conditions_particles=['absorbing', 'absorbing'],
+    warpx_max_grid_size=max_grid_size,
+    warpx_blocking_factor=blocking_factor)
+
+# Particles: plasma parameters
+# critical plasma density
+nc = 1.742e27  # [m^-3]  1.11485e21 * 1.e6 / 0.8**2
+# number density: "fully ionized" electron density as reference
+#   [material 1] cryogenic H2
+n0 = 30.0  # [n_c]
+#   [material 2] liquid crystal
+# n0 = 192
+#   [material 3] PMMA
+# n0 = 230
+#   [material 4] Copper (ion density: 8.49e28/m^3; times ionization level)
+# n0 = 1400
+plasma_density = n0 * nc
+preplasma_L = 0.05e-6  # [m] scale length (>0)
+preplasma_Lcut = 2.0e-6  # [m] hard cutoff from surface
+plasma_r0 = 2.5e-6  # [m] radius or half-thickness
+plasma_eps_z = 0.05e-6  # [m] small offset in z to make zmin, zmax interval larger than 2*(r0 + Lcut)
+plasma_creation_limit_z = plasma_r0 + preplasma_Lcut + plasma_eps_z  # [m] upper limit in z for particle creation
+
+plasma_xmin = None
+plasma_ymin = None
+plasma_zmin = -plasma_creation_limit_z
+plasma_xmax = None
+plasma_ymax = None
+plasma_zmax = plasma_creation_limit_z
+
+slab_with_ramp_dist = picmi.AnalyticDistribution(
+    density_expression='plasma_density*((abs(z)<=r0) + (abs(z)<r0+Lcut) * (abs(z)>r0) * np.exp(-(abs(z)-r0)/L))',
+    lower_bound=[plasma_xmin, plasma_ymin, plasma_zmin],
+    upper_bound=[plasma_xmax, plasma_ymax, plasma_zmax],
+    # the following keyword arguments are setting shortcuts for the constants defined above
+    r0=plasma_r0,
+    L=preplasma_L,
+    Lcut=preplasma_Lcut
```

Comment:
If you like, you can also use f strings:
```suggestion
    density_expression=f"{plasma_density}*((abs(z)<={plasma_r0}) + (abs(z)<{plasma_r0}+{preplasma_Lcut}) * (abs(z)>{preplasma_L}) * np.exp(-(abs(z)-{plasma_r0})/{preplasma_L}))",
    lower_bound=[plasma_xmin, plasma_ymin, plasma_zmin],
    upper_bound=[plasma_xmax, plasma_ymax, plasma_zmax]
```

Reply:
Thanks! this was actually necessary later. The uploaded version was outdated